### PR TITLE
Fixes #14604 - Allow content_view_version#index search on CVV names

### PR DESCRIPTION
**Pull request by akofink**
Corequisite: https://github.com/Katello/hammer-cli-katello/pull/386

### CODE REVIEW

File: app/controllers/katello/api/v2/content_view_versions_controller.rb [hunk 1]
```diff
@@ -33,6 +33,7 @@ def index_relation
       versions = versions.in_environment(@environment) if @environment
       versions = versions.with_puppet_module(@puppet_module) if @puppet_module
       versions = versions.component_of(params[:composite_version_id]) if params[:composite_version_id]
+      versions = versions.with_names(JSON.parse(params[:names])) if params[:names]
```
Comment (daviddavis) [inline app/controllers/katello/api/v2/content_view_versions_controller.rb]: Calling `JSON.parse` here seems odd. Either you're expecting a JSON encoded string within your POST request (probably should use an array instead) or in your JSON request (probably shouldn't send JSON inside JSON).

File: app/controllers/katello/api/v2/content_views_controller.rb [hunk 2]
```diff
@@ -45,6 +45,7 @@ def index_relation
       content_views = content_views.non_composite if params[:noncomposite]
       content_views = content_views.composite if params[:composite]
       content_views = content_views.where(:name => params[:name]) if params[:name]
+      content_views = content_views.where(name: JSON.parse(params[:names])) if params[:names]
```
Comment (daviddavis) [inline app/controllers/katello/api/v2/content_views_controller.rb]: The `name` parameter should already accept an array of names.

**Comment by daviddavis** [comment]
Your commit/PR name needs a `#`.

File: app/services/katello/database_agnostic_helper.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+module Katello
+  module DatabaseAgnosticHelper
+    # Methods specific to a database
+    module PostgreSQL
+      def concat(*args)
+        "(#{args.join(' || ')})"
+      end
+
+      def quote(s)
```
Comment (daviddavis) [inline app/services/katello/database_agnostic_helper.rb]: There should already be a quote method in Rails I think.

**Comment by ehelms** [comment]
The issue reads like the user wants to be able to add content views by name, not content view versions by name. What am I missing that requires us to add this searching by content view version "name" which is a contrived field that combines the name of the content view and version of the content view version?

**Comment by akofink** [comment]
@ehelms I discussed this in foreman-dev before I started, and this is what we agreed upon. I think the customer who created the bug didn't quite realize the difference between content views (CVs) and content view versions (CVVs). That said, the bug states that they want to create _composite_ content views using names for the components. Since the components are technically CVVs, which have only a computed name rather than a name field in the DB, we decided to follow the format of CVV names the user sees when issuing `hammer content-view version list --organization-id 1`. I can change the implementation or even scratch it if you think it shouldn't be a feature.

```
$ hammer content-view version list --organization-id 1
---|-------------------------------|---------|-----------------------
ID | NAME                          | VERSION | LIFECYCLE ENVIRONMENTS
---|-------------------------------|---------|-----------------------
5  | cv2 3.0                       | 3.0     | Library               
4  | cv2 2.0                       | 2.0     |                       
3  | cv2 1.0                       | 1.0     |                       
2  | cv1 1.0                       | 1.0     | Library               
1  | Default Organization View 1.0 | 1.0     | Library               
---|-------------------------------|---------|-----------------------
```

**Comment by daviddavis** [comment]
That sucks that the content view info [was removed here](https://github.com/Katello/hammer-cli-katello/commit/8958e24eb5486c978c6cbdf172c6a04376e2bf43). It kinda makes it look like there's an actual name field on versions when there's not.

**Comment by daviddavis** [comment]
For one thing, I worry that when a user updates a content view name, they might be surprised to see that their content view version names also got automatically updated somehow. Anyway, this might be a separate discussion so I opened http://projects.theforeman.org/issues/14884.

**Comment by daviddavis** [comment]
Digging a little more into this a little more, it looks like we have code that was maybe intended to support adding versions by name:

https://github.com/Katello/hammer-cli-katello/blob/master/lib/hammer_cli_katello/content_view.rb#L210-L238

It could probably be built out to have options like content view name, version (major.minor).

**Comment by akofink** [comment]
So are we agreed that CVVs should not be able to be searched for with the computed `name` attribute? If not, I'll push back on this bug and close this PR. As a side note, version is also a computed field (`major.minor`).

**Comment by daviddavis** [comment]
Yes, IMO I would ditch the name option idea on the condition that CLI users can add a version to a composite with content view name, minor, and major options.

**Pull request closed without merging**